master/buildbot/schedulers/timed.py

```diff
@@ -206,30 +206,31 @@ def now(self):
         "Similar to util.now, but patchable by tests"
         return util.now(self._reactor)
 
+    @defer.inlineCallbacks
     def _scheduleNextBuild_locked(self):
         # clear out the existing timer
         if self.actuateAtTimer:
             self.actuateAtTimer.cancel()
         self.actuateAtTimer = None
 
         # calculate the new time
-        d = self.getNextBuildTime(self.lastActuated)
+        actuateAt = yield self.getNextBuildTime(self.lastActuated)
 
-        # set up the new timer
-        @d.addCallback
-        def set_timer(actuateAt):
+        if actuateAt is None:
+            self.actuateAt = None
+        else:
+            # set up the new timer
             now = self.now()
             self.actuateAt = max(actuateAt, now)
```

Comment:
Note: I have also modified this part, where getNextBuildTime can return None (according to it's docs) and it would be passed to max fce. This would result in error.